src/exactlyOnceTest/java/com/clickhouse/ExactlyOnceTest.java

```diff
@@ -150,32 +150,35 @@ public static void tearDown() throws IOException, URISyntaxException, Interrupte
     }
 
 
-    private void compareCounts(String topicName) throws InterruptedException {
+    private boolean compareCounts(String topicName) throws InterruptedException {
         ClickHouseResponse clickHouseResponse = clickhouseAPI.clearTable(topicName);
         LOGGER.info(String.valueOf(clickHouseResponse));
         clickHouseResponse.close();
 
         Integer count = sendDataToTopic(topicName);
+        LOGGER.info("Expected Total: {}", count);
+        waitForCount(topicName);
 
-        Thread.sleep(60 * 1000);
-        LOGGER.info("Actual Total: {}", count);
-        int[] counts = clickhouseAPI.count(topicName);
-        if (counts != null) {
-            assertEquals(count, counts[1]);
+        int[] databaseCounts = clickhouseAPI.count(topicName);
```

Comment:
Sorry for being so picky, but why return an array of ints and not Optional int  
This way, you can check if it has value and then extract it https://www.baeldung.com/java-optional